modules/auxiliary/cloud/aws/enum_ec2.rb

```diff
@@ -31,15 +40,9 @@ def initialize(info = {})
     )
   end
 
-  def handle_aws_errors(e)
-    if e.class.module_parents.include?(Aws)
-      fail_with(Failure::UnexpectedReply, e.message)
-    else
-      raise e
-    end
-  end
-
   def enumerate_regions
+    return [datastore['REGION']] if datastore['REGION']
+
```

Comment:
I think this change should be reverted. It's misleading to call the method `enumerate_regions` and then not enumerate the regions. It made more sense the way it was, that is to check if you have a specific region and then to enumerate them if not.